daemon.c

```diff
@@ -1540,8 +1556,10 @@ void *receiveOOK(void *param) {
 					/* Let's do a little filtering here as well */
 					if(r.length >= hw->minrawlen && r.length <= hw->maxrawlen) {
 						receive_queue(r.pulses, r.length, plslen, hw->hwtype);
+						r.length = 0;
 					}
-					r.length = 0;
+					// Avoid that short footer pulses (X10, IMPULS, ...) interfere with long header /sync pulses
+					if(r.length>20)r.length = 0;	// Avoid a buffer reset and preserve potential Header / SYNC / Start pulses
```

Comment:
I think the approach is okay and valid, previously the counter was always reset.

The current logic is based on s*mfy requirements, it is compatible with all protocols using footer detection.

There is a certain probability that the 1st pulsetrain is now discarded, due to a partially filled buffer, but that has not proven to be a problem in daily use, as most pulsetrains are repeated several times.

Von: CurlyMoo [mailto:notifications@github.com] 
Gesendet: Sonntag, 20. Dezember 2015 19:35
An: pilight/pilight
Cc: wo-rasp
Betreff: Re: [pilight] S*mfy Protocol (#200)

 

In daemon.c <https://github.com/pilight/pilight/pull/200#discussion_r48106573> :

>                                     }
> -                                   r.length = 0;
> +                                   // Avoid that short footer pulses (X10, IMPULS, ...) interfere with long header /sync pulses
> +                                   if(r.length>20)r.length = 0;   // Avoid a buffer reset and preserve potential Header / SYNC / Start pulses

In the past this problem was neglected, the first pulsetrain was discarded, the header from the second (third, forth, …) pulsetrain was interpreted as footer, and the last pulsetrain was discarded.

I'm just not sure if this is the right solution. I believe we should create some logic to properly detect those headers instead of just dumb resetting the counter.

—
Reply to this email directly or view it on GitHub <https://github.com/pilight/pilight/pull/200/files#r48106573> .  <https://github.com/notifications/beacon/AG9MV6Uqitg3wz6tdsh9rTHz57jwCpzlks5pRuxNgaJpZM4GUJeH.gif>